lms/djangoapps/discussion_api/api.py

```diff
@@ -39,6 +41,33 @@ def _get_course_or_404(course_key, user):
     return course
 
 
+def _get_thread_and_context(request, thread_id, parent_id=None, retrieve_kwargs=None):
+    """
+    Retrieve the given thread and build a serializer context for it, returning
+    both. This function also enforces access control for the thread (checking
+    both the user's access to the course and to the thread's cohort if
+    applicable). Raises Http404 if the thread does not exist or the user cannot
+    access it.
+    """
+    retrieve_kwargs = retrieve_kwargs or {}
+    try:
+        if "mark_as_read" not in retrieve_kwargs:
+            retrieve_kwargs["mark_as_read"] = False
+        cc_thread = Thread(id=thread_id).retrieve(**retrieve_kwargs)
+        course_key = CourseLocator.from_string(cc_thread["course_id"])
+        course = _get_course_or_404(course_key, request.user)
+        context = get_context(course, request, cc_thread, parent_id)
+        if not context["is_requester_privileged"] and cc_thread["group_id"]:
+            requester_cohort = get_cohort_id(request.user, course.id)
+            if requester_cohort is not None and cc_thread["group_id"] != requester_cohort:
```

Comment:
Actually, only 1 is true, because get_cohort_id returns None if the course is not currently cohorted, even if the user has been assigned a cohort in the course. I will double check with product whether 2 is acceptable; if not, it should be a fairly easy change.